File: code-coaches/easy/izzy-the-iguana/IzzyTheIguana.swift
import Foundation

func getSnackPoints(_ snacks: [String]) -> Int {
    let snacksDict = [
        "Mango": 9,
        "Lettuce": 5,
        "Carrot": 4,
        "Cheeseburger": 0,
    ]
    let points = snacks.reduce(0) { $0 + (snacksDict[$1] ?? 0) }
    return points
}

func main() {
    guard let line = readLine() else { return }

    let snacks = line.components(separatedBy: .whitespacesAndNewlines)
    if getSnackPoints(snacks) >= 10 {
        print("Come on Down!")
    } else {
        print("Time to wait")
    }
}

main()

File: code-coaches/easy/izzy-the-iguana/IzzyTheIguana.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class IzzyTheIguana
{
    static int GetSnackPoints(List<string> snacks)
    {
        Dictionary<string, int> snacksDict = new Dictionary<string, int>
        {
            { "Mango", 9 },
            { "Lettuce", 5 },
            { "Carrot", 4 },
            { "Cheeseburger", 0 }
        };
        int points = snacks.Sum(snack => snacksDict.GetValueOrDefault(snack, 0));
        return points;
    }

    static void Main(string[] args)
    {
        List<string> snacks = Console.ReadLine().Split().ToList();
        if (GetSnackPoints(snacks) >= 10)
            Console.WriteLine("Come on Down!");
        else
            Console.WriteLine("Time to wait");
    }
}
